src/Components/Profile/ProfileSidebar.tsx

```diff
@@ -1,11 +1,83 @@
 import React from 'react'
-import { getJSDocTypeParameterTags } from 'typescript'
+import { CourseBlock, CourseBlockWeek, parseCookie } from '../../modules/API'
 import contexts from '../APIContext'
 
+interface DisplayBlock extends CourseBlock {
+  days: ('M' | 'T' | 'W' | 'R' | 'F')[]
+}
+
 export const ProfileSidebar = () => {
+  const findScheduled = (week: CourseBlockWeek) => {
+    const id = parseCookie().tias_user_id;
+    let retData: CourseBlock[] = [];
+
+    const days = [week.Monday, week.Tuesday, week.Wednesday, week.Thursday, week.Friday];
+    days.forEach(day => {
+      day?.forEach(block => {
+        if (block.scheduled?.includes(+id)) retData.push(block)
+      })
+    })
+
+    return retData;
+  }
+
+  const formatDate = (date: Date) => {
+    const hour = date.getHours();
+    const minute = date.getMinutes();
+    const ampm = hour >= 12 ? 'pm' : 'am';
+    const hour12 = (hour === 12) ? 12 : hour % 12;
+    const minutes = minute < 10 ? `0${minute}` : minute;
+    return `${hour12}:${minutes} ${ampm}`;
+  }
+
+  const renderScheduled = (retData: CourseBlock[]) => {    
+    let retFormat: DisplayBlock[] = [];
+    let ret: JSX.Element[] = [];
+
+    const dayMap = new Map(
+      [
+        ['Monday', 'M'],
+        ['Tuesday', 'T'],
+        ['Wednesday', 'W'],
+        ['Thursday', 'R'],
+        ['Friday', 'F']
+      ]
+    )
+
+    retData.sort((a, b) => {
+      if (a.course_number < b.course_number) return -1;
+      else if (a.course_number > b.course_number) return 1;
+      else if (a.section_number < b.section_number) return -1;
+      else if (a.section_number > b.section_number) return 1;
+      return 0;
+    }).forEach(block => {
+      const where = retFormat.findIndex(b => b.course_number === block.course_number && b.section_number === block.section_number);
+      if (where === -1) retFormat.push({...block, days: [dayMap.get(block.weekday) as 'M' | 'T' | 'W' | 'R' | 'F']})
+      else retFormat[where].days.push(dayMap.get(block.weekday) as 'M' | 'T' | 'W' | 'R' | 'F');
+    })
+
+    retFormat.forEach(block => {
+      ret.push(
+        <div className="schedule-info-element" key={`sie-${JSON.stringify(block)}`}>
+          <div>
+            {block.department}: {block.course_number}-{block.section_number}
+          </div>
+          <div>
+            {formatDate(block.start_time)}-{formatDate(block.end_time)} {block.days.join('')}
+          </div>
+          <div>
+            {block.place}
+          </div>
+        </div>
+      )
+    })
```

Comment:
Does this fill the sidebar below the name with the assigned courses?
![image](https://user-images.githubusercontent.com/15470694/161401700-f97b85a8-c648-47ea-8d62-9f1090eaa227.png)
It seems empty, so if this doesn't do that, can we consider doing that? In general, how can I see what this is doing?

Reply:
Try generating a schedule and checking back to that page